### PR TITLE
feat(session-start): make instinct injection count and confidence threshold configurable

### PR DESCRIPTION
**Pull request by gaurav0107**
## What Changed

Makes the two SessionStart instinct-injection knobs in
`scripts/hooks/session-start.js` configurable via environment, keeping the
current values as defaults:

- **`ECC_MAX_INJECTED_INSTINCTS`** — cap on how many learned instincts get
  injected into context at SessionStart (default `6`).
- **`ECC_INSTINCT_CONFIDENCE_THRESHOLD`** — minimum confidence, in `[0, 1]`,
  an instinct needs to be injected (default `0.7`).

The two hardcoded constants (`MAX_INJECTED_INSTINCTS`,
`INSTINCT_CONFIDENCE_THRESHOLD`) become `DEFAULT_`-prefixed fallbacks and are
now resolved through `getMaxInjectedInstincts()` /
`getInstinctConfidenceThreshold()`, mirroring the file's existing
`getSessionRetentionDays()` / `getSessionStartMaxContextChars()` env-override
pattern. Invalid or out-of-range values fall back to the defaults.

This is **part (a)** of #2371 — the smaller, self-contained change the issue
author offered to land first ("Happy to scope it to just (a) first if you
prefer the smaller change"). Part (b) (confidence + stack/project relevance
ranking) is intentionally left for a separate PR to keep this one surgical.

## Why This Change

Instinct selection is currently confidence-only with a hardcoded count and
threshold, so tuning injection behavior requires editing (and re-editing after
every update) source constants. Exposing them as env vars removes that friction
— e.g. lower `ECC_MAX_INJECTED_INSTINCTS` for low-context/local-model setups, or
raise `ECC_INSTINCT_CONFIDENCE_THRESHOLD` to inject only high-confidence
instincts — with zero behavior change when the vars are unset.

## Testing Done

- [x] `node tests/run-all.js` — **2939/2939 pass** (clean env, `CLAUDE_PLUGIN_ROOT` unset)
- [x] Added `tests/hooks/hooks.test.js` subprocess coverage:
  - default cap injects 6; `ECC_MAX_INJECTED_INSTINCTS=3` injects 3; garbage value falls back to 6
  - default `0.7` threshold injects only above-floor instincts; `0.95` filters all out; `0.5` passes all (still capped at 6)
- [x] `npx eslint scripts/hooks/session-start.js tests/hooks/hooks.test.js` — clean
- [x] `npx markdownlint README.md` — clean
- [x] `scripts/ci/validate-no-personal-paths.js`, `check-unicode-safety.js`, `validate-hooks.js` — all pass

## Type of Change

- [x] New feature (non-breaking; env-gated, defaults preserved)

## Security & Quality Checklist

- [x] No new dependencies
- [x] No secrets, no network calls, no personal absolute paths
- [x] Env values validated (`Number.parseInt` / `Number.parseFloat` + range checks) with safe fallback
- [x] Surgical scope — one concern, single production file + its test + a doc note
- [x] Behavior unchanged when the new env vars are unset

## Documentation

- README "Hook Runtime Controls" section documents both new variables in the
  existing comment+export style.

Part of #2371 — implements part (a) (env-configurable knobs). Part (b)
(confidence + stack/project relevance ranking) is left as a follow-up; happy
to open that separately or fold it in if you'd prefer a single PR.

## Summary by CodeRabbit

* **New Features**
  * Added environment-based controls for learned instinct injection at session start, including a configurable maximum count and confidence threshold.

* **Bug Fixes**
  * More reliable handling of invalid environment values by falling back to safe defaults.
  * Injection behavior now consistently applies both the confidence threshold and the maximum injection cap.

* **Documentation**
  * Updated README with the new environment variables and their default values.

* **Tests**
  * Added edge-case tests covering defaulting, invalid overrides, threshold filtering, and max-limit enforcement.

### CODE REVIEW

File: scripts/hooks/session-start.js [hunk 1]
```diff
@@ -116,6 +116,40 @@ function getSessionStartMaxContextChars() {
...
+ *
+ * @returns {number} The cap on injected instincts.
+ */
+function getMaxInjectedInstincts() {
+  const raw = process.env.ECC_MAX_INJECTED_INSTINCTS;
+  if (!raw) return DEFAULT_MAX_INJECTED_INSTINCTS;
+
+  const parsed = Number.parseInt(raw, 10);
+  return Number.isInteger(parsed) && parsed > 0 ? parsed : DEFAULT_MAX_INJECTED_INSTINCTS;
+}
```
[the remark this answers is not in the document]
Comment (gaurav0107) [reply scripts/hooks/session-start.js:163]: Good catch. Switched both resolvers to Number() after a trim (with an empty-string guard), so partial values like 3.9, 6abc, or 0.7x now fall back to the default instead of accepting the numeric prefix. Added a regression assertion for the non-integer count. Addressed in 297eb29d.

File: scripts/hooks/session-start.js [hunk 2]
```diff
@@ -116,6 +116,52 @@ function getSessionStartMaxContextChars() {
...
+  const raw = process.env.ECC_INSTINCT_CONFIDENCE_THRESHOLD;
+  if (!raw) return DEFAULT_INSTINCT_CONFIDENCE_THRESHOLD;
+
+  // Use Number() (not parseFloat) so a malformed value like "0.7x" or
+  // "0.7 " with trailing junk is rejected whole rather than silently
+  // accepting the numeric prefix.
+  const normalized = raw.trim();
+  if (!normalized) return DEFAULT_INSTINCT_CONFIDENCE_THRESHOLD;
+
+  const parsed = Number(normalized);
```
[the remark this answers is not in the document]
Comment (gaurav0107) [reply scripts/hooks/session-start.js:137]: Thanks — tightened both resolvers to validate the format before converting: /^\d+(\.\d+)?$/ for the 0-1 threshold and /^\d+$/ for the positive-integer count, so 0x1, 1e2, and other non-decimal or partial syntax now fall back to the default. Added regression assertions for 1e2 and 0x1. Addressed in 1b79e77f.

**Review by daltino — APPROVED** [review]
This PR introduces configurable parameters for the instinct injection count (`ECC_MAX_INJECTED_INSTINCTS`) and confidence threshold (`ECC_INSTINCT_CONFIDENCE_THRESHOLD`), which enhances flexibility for the `SessionStart` feature. The changes align with the contribution guidelines, as they add valuable configuration options to hooks, include documentation updates in `README.md`, and come with appropriate tests verifying the feature's behavior. The changes are well-documented and maintain backward compatibility with the default settings.

**Comment by gaurav0107** [comment]
Quick note: these CI reds aren't from this PR. `main` itself is failing because its lockfile is out of sync with the eslint 10 bump, so `npm ci` errors before any test runs. #2402 fixes that by resyncing the lockfiles. The bun/pnpm jobs here (including this PR's new tests) are green.

The leftover Windows and Python Tests reds are separate, already-existing issues (a bash-detection test and a missing pyyaml), not related to this change.
